doc/pic/hld_fpmsyncd.md

```diff
@@ -230,14 +235,23 @@ admin@sonic:~$ sonic-db-cli APPL_DB HGETALL ROUTE_TABLE:10.1.1.4
   <tr><td>NHA_GROUP</td><td>[{125,1},{126,1}]</td></tr>
   <tr><td rowspan="2">4</td><td rowspan="2">RTM_NEWROUTE</td><td>RTA_DST</td><td>10.1.1.4</td></tr>
   <tr><td>RTA_NH_ID</td><td>ID127</td></tr>
+  <tr><td rowspan="2">5</td><td rowspan="2">RTM_NEWROUTE</td><td>RTA_DST</td><td>10.1.1.5</td></tr>
+  <tr><td>RTA_NH_ID</td><td>ID127</td></tr>
+  <tr><td rowspan="3">6</td><td rowspan="3">RTM_NEWNEXTHOP</td><td>NHA_ID</td><td>ID128</td></tr>
+  <tr><td>NHA_GATEWAY</td><td>10.0.0.4</td></tr>
+  <tr><td>NHA_OIF</td><td>22</td></tr>
+  <tr><td rowspan="2">7</td><td rowspan="2">RTM_NEWNEXTHOP</td><td>NHA_ID</td><td>ID127</td></tr>
+  <tr><td>NHA_GROUP</td><td>[{125,1},{128,1}]</td></tr>
 </table>
 
 A short description of `fpmsyncd` logic flow:
 
-- When receiving `RTM_NEWNEXTHOP` events on sequence 1, 2 and 3, `fpmsyncd` will save the information in an internal list to be used when necessary.
-- When `fpmsyncd` receive the `RTM_NEWROUTE` on sequence 4, the process will write the NextHop Group with ID 118 to the `NEXTHOP_GROUP_TABLE` using the information of gateway and interface from the NextHop Group events with IDs 116 and 117.
-- Then `fpmsyncd` will create a new route entry to `ROUTE_TABLE` with a `nexthop_group` field with value `ID118`.
-- When `fpmsyncd` receives the last `RTM_NEWROUTE` on sequence 5, the process will create a new route entry (but no NextHop Group entry) in `ROUTE_TABLE` with `nexthop_group` field with value `ID118`. (Note: This NextHop Group entry was created when the `fpmsyncd` received the event sequence 4.)
+- When receiving `RTM_NEWNEXTHOP` events on sequence 1 and 2, `fpmsyncd` will write the information in `NEXTHOP_GROUP_TABLE` with NextHop Group `ID125` and `ID126` and with information of gateway and interface respectively. For sequence 3  multi path NextHop Group `ID127` the information in `NEXTHOP_GROUP_TABLE` will contain the list of IDs 125 and 126
```

Comment:
What changes are expected towards kernel programming? please capture those details if not captured in a separate document.

Reply:
No new changes are required from frr/kernel side for this feature.  When NHG is enabled in FRR, it already programs routes to kernel using reference to the NHID.  This implementation in fact make the FPM programming similar to how the routes are being programmed in kernel already